tests/test_user_security.py

```diff
@@ -0,0 +1,370 @@
+    allowed_update_types = [
+        ValueRuleValidator('username', str),
+        ValueRuleValidator('email', str),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, User, connect_to_mem_db(), UserDataAccess.rules_validator)
+
+
+# CRUD INIT
+user_data_access = UserDataAccess()
+
+
+@Cache(key='user_data_{u_id}', expire=timedelta(seconds=30))
+def get_user(u_id):
+    return result_to_dict(user_data_access.get(u_id))
+
+
+# USER SECURITY SETUP
+class SessionUser(object):
+    def __init__(self, id: int, email: str, role: int, status: int):
+        self.id = id
+        self.email = email
+        self.role = role
+        self.status = status
+
+    def __dict__(self):
+        return {'id': self.id, 'email': self.email, 'role': self.role, 'status': self.status}
+
+
+def has_access(user_session, check_policies):
+    user_role = user_session.role
+    role = []
+    status = User.Status.ACTIVE.value
+    user_status = user_session.status
+    for policy in check_policies:
+        if type(policy) == User.Status:
+            status = policy.value
+        if type(policy) == User.PolicyRoles:
+            role.append(policy.value)
+    if not role:
+        role.append(User.PolicyRoles.USER.value)
+    if user_role <= max(role) and user_status == status:
+        return True
+    else:
+        return False
+
+
+class CustomerSecurity(UserSecurity):
+    def __init__(self, cookie_name: str, secret: str, expiration_time: timedelta):
+        UserSecurity.__init__(self, cookie_name, JWTTokenHandler(secret, expiration_time))
+
+    def secure_entry(self, request, session_obj: SessionUser, policies: list):
+        if not policies:
+            return response_status(HTTPStatus.OK)
+        elif has_access(session_obj, policies):
+            return response_status(HTTPStatus.OK)
+        else:
+            return response_status(HTTPStatus.UNAUTHORIZED)
+
+    def from_session_data(self, session_data: dict) -> SessionUser:
+        return SessionUser(session_data['id'], session_data['email'], session_data['role'], session_data['status'])
+
+    def generate_session_data(self, user) -> dict:
+        return {
+            'id': user['id'],
+            'email': user['email'],
+        }
+
+
+# SECURITY HANDLER REGISTER
+secret_key = 'super-secret'
+security_handler = CustomerSecurity('user_cookie', secret_key, timedelta(seconds=30))
+SecurityHandler.register(security_handler)
+
+# DJANGO INIT
+if not settings.configured:
+    settings.configure()
+    settings.DEFAULT_CHARSET = 'utf-8'
+
+web_util = WebHelpersUtils()
+web_util.init(web_util.ServerType.DJANGO)
+
+
+class TestUserSecurity(unittest.TestCase):
+    @classmethod
+    def setUp(cls):
+        CoreLib.cache_registry.unregister('test_user_security')
+
+        cls.admin = result_to_dict(
+            user_data_access.create({'username': 'admin', 'email': 'admin@def.com', 'role': User.PolicyRoles.ADMIN})

```

Comment:
if all users' status is ACTIVE. please specify it here so it will be clear. 
also, let's add 2 users with different status for the test